eslint-rules/lib/rules/prop-value-shape-deprecation.js

```diff
@@ -0,0 +1,141 @@
+                  spreadSource.properties[0].value.properties,
+                  spreadSourceName,
+                  deprecation,
+                  nodeName,
+                  node,
+                  context
+                );
+              } else if (_.includes(deprecation.propNames, attributeName)) {
+                const attributeType = _.get(attribute, 'value.expression.type');
+                if (attributeType === 'Identifier') {
+                  const passedProp = utils.findValueNodeOfIdentifier(
+                    attribute.value.expression.name,
+                    context.getScope()
+                  );
+                  if (passedProp && passedProp.properties) {
+                    checkAttributeProperties(
+                      passedProp.properties,
+                      attributeName,
+                      deprecation,
+                      nodeName,
+                      node,
+                      context
+                    );
+                  }
+                }
+                const attributeProps = _.get(attribute, 'value.expression.properties');
+                for (let index = 0; index < attributeProps.length; index++) {
+                  const spreadElementType = _.get(attribute, `value.expression.properties[${index}].type`);
+                  if (attributeType === 'ObjectExpression' && spreadElementType === 'ExperimentalSpreadProperty') {
+                    const spreadSource = utils.findValueNodeOfIdentifier(
+                      attribute.value.expression.properties[index].argument.name,
+                      context.getScope()
+                    );
+                    if (spreadSource && spreadSource.properties) {
+                      checkAttributeProperties(spreadSource.properties, attributeName, deprecation, nodeName, node, context);
+                    }
+                  }
+                }
+                const attributeProperties = _.get(attribute, 'value.expression.properties');
+                checkAttributeProperties(attributeProperties, attributeName, deprecation, nodeName, node, context);
+              }
+            });
+          }
+        });
+      } catch (err) {
+        console.log('Found error in: ', context.getFilename());
+      }
+    }
+
+    function checkAttributeProperties(attributeProperties, attributeName, deprecation, nodeName, node, context) {
+      for (let i = 0; i <= attributeProperties.length; i++) {
+        const propertyName = _.get(attributeProperties[i], 'key.name');
+        const origin = propertyName && _.find(deprecation.shape, ['prop', propertyName]);
+        if (origin && origin.prop && propertyName === origin.prop) {
+          const destination = _.get(origin, 'fix.propName');
+          const message = `The shape of '${attributeName}' prop of '${nodeName}' doesn't contain '${origin.prop}' anymore. Please use '${destination}' instead (fix is available).`;
+          context.report({
```

Comment:
I'm not sure if generally it's expected the report to be first (in a separate function), but it seems that's the convention we have in our rules.
Please look at our other rules and apply the same structure so it'll be easier to compare.